contracts/rewards/src/contract/execute.rs

```diff
@@ -77,113 +83,110 @@ where
     ) -> Result<(), ContractError> {
         let cur_epoch = self.current_epoch(block_height)?;
 
+        let event =
+            self.load_or_store_event(event_id, contract_addr.clone(), cur_epoch.epoch_num)?;
+
+        self.store
+            .load_epoch_tally(contract_addr.clone(), event.epoch_num)?
+            .unwrap_or(EpochTally::new(
+                contract_addr,
+                cur_epoch,
+                self.store.load_params().params,
+            ))
+            .record_participation(worker)
+            .then(|mut tally| {
+                if matches!(event, LoadState::New(_)) {
+                    tally.event_count += 1
+                }
+                self.store.save_epoch_tally(&tally)
+            })
+    }
+
+    fn load_or_store_event(
+        &mut self,
+        event_id: nonempty::String,
+        contract_addr: Addr,
+        cur_epoch_num: u64,
+    ) -> Result<LoadState<Event>, ContractError> {
         let event = self
             .store
             .load_event(event_id.to_string(), contract_addr.clone())?;
 
-        if event.is_none() {
-            self.store.save_event(&Event::new(
-                event_id,
-                contract_addr.clone(),
-                cur_epoch.epoch_num,
-            ))?;
-        }
-
-        let tally = match event {
-            Some(event) => self
-                .store
-                .load_epoch_tally(contract_addr.clone(), event.epoch_num)?
-                .expect("couldn't find epoch tally for existing event"),
-            None => self
-                .store
-                .load_epoch_tally(contract_addr.clone(), cur_epoch.epoch_num)?
-                .unwrap_or(EpochTally::new(
-                    contract_addr,
-                    cur_epoch,
-                    self.store.load_params().params,
-                )) // first event in this epoch
-                .increment_event_count(),
+        match event {
+            None => {
+                let event = Event::new(event_id, contract_addr, cur_epoch_num);
+                self.store.save_event(&event)?;
+                Ok(LoadState::New(event))
+            }
+            Some(event) => Ok(LoadState::Loaded(event)),
         }
-        .record_participation(worker);
-
-        self.store.save_epoch_tally(&tally)?;
-
-        Ok(())
     }
 
     pub fn distribute_rewards(
         &mut self,
-        contract: Addr,
+        contract_addr: Addr,
```

Comment:
I find the `_addr` suffix is pretty redudant. The type is already `Addr`.

Reply:
I renamed it to `target_contract`, I want to avoid confusion, because self is of type `Contract`